src/tactic/command/global_search_trigger.py

```diff
@@ -341,12 +354,24 @@ def update_collection_keywords(my, mode, base_search_type, input):
         parent_sobject = Search.get_by_code(asset_stype, parent_code)
         child_sobject = Search.get_by_code(asset_stype, search_code)
         
-
         collection_keywords_dict = {}
         parent_collection_keywords_dict = {}
 
         # Existing "collection" keywords in child's keywords_data
         child_keywords_data = child_sobject.get_json_value("keywords_data", {})
+        if isinstance(child_keywords_data, basestring):
+            raise TacticException("Invalid data found in keywords_data for %s. Please notify site administrator to correct it."%child_sobject.get_code())
+
+       
+        # check for old data structure with only keywords filled and initialize if necessary
+        if not child_keywords_data:
+            user_keywords = child_sobject.get_value('user_keywords')
+            original_keywords = child_sobject.get_value('keywords')
+            if original_keywords and not user_keywords:
+                # initiatize keywords_data in this case
+                child_keywords_data['user'] = original_keywords
```

Comment:
We may not need this since the trigger takes care of entering it into keywords_data when the user_keywords are changed